hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/EthereumTransactionHandler.java

```diff
@@ -77,12 +94,23 @@ public void handle(@NonNull final HandleContext context) throws HandleException
         // Create the transaction-scoped component
         final var component = provider.get().create(context, ETHEREUM_TRANSACTION);
 
+        // Assemble the appropriate top-level record for the result
```

Comment:
⚠️ Hmm none of these changes should be necessary, since the `MessageCallProcessor` should (must!) refuse to lazy-create a long-zero address. 🤔

Is the only problem that you get e.g. `CONTRACT_REVERT_EXECUTED` instead of `INVALID_CONTRACT_ID` when enabling these `@HapiTest`s?